core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -194,12 +194,38 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv, numUsableCores: Int) exte
     endpoints.containsKey(name)
   }
 
+  def getNumOfThreads(conf: SparkConf): Int = {
+    // try to get specific threads configurations of driver and executor
+    val executorId = conf.get("spark.executor.id", "")
+    // neither driver nor executor if executor id is not set
+    var role = ""
```

Comment:
we can use `val role = if (...)` or `val role = executorId match ...` here.

Reply:
re-structured code as discussed. thanks